ecosystem/sep-0008.md

```diff
@@ -61,107 +72,211 @@ approval_server="https://goat.io/tx_approve"
+An `action_required` response will have a `200` HTTP status code and `action_required` as the `status` value.
 
 Parameters:
 
-Name | Data Type | Description
------|-----------|------------
-status|string|`"rejected"`
-error|string|A human readable string explaining why the transaction is not compliant and could not be made compliant.
+Name | Type | Description
+-----|------|------------
+`status` | string | `"action_required"`
+`message` | string | A human readable string containing information regarding the action required.
+`action_url` | string | A URL that allows the user to complete the actions required to have the transaction approved.
 
-### Best practices
+###### Example
 
-- If a transaction was revised, ALWAYS explain the changes that were made to a transaction through the `message` parameter.
-- Core operations shouldn't be modified as that can be confusing and misleading. For example, if the user wishes to put an offer for 1000 GOATS but due to velocity limits they can only put an offer for 500 GOATS, it is better to error with a message than to change the amount.
-- Adding an upper timebound to a transaction can help the issuer ensure that their view of the world does not get out of sync.
-- Issuers can enforce additional fees by adding additional operations. For example, any transaction involving goats, will also send 0.1 GOAT to Boris’ account.
-- Once a transaction has been signed, the issuer should mark it as pending and take it into account, as long as it hasn’t been timed out, so that they can have a accurate view of the world.
+```json
+{
+  "status": "action_required",
+  "message": "Please sign the terms of service via the provided URL.",
+  "action_url": "https://..."
+}
+```
 
-## Account Setup
+##### Rejected
 
-Implementing Regulated Assets requires the participating accounts to be “managed” by the issuer. This is achieved by having the issuer be a co-signer on the account, such that a medium threshold operation cannot be submitted without the issuer’s approval.
+This response means that the transaction is not compliant and could not be revised to be made compliant.
 
-In the future, this requirement can be replaced by introducing protocol level [CoSigned assets](https://github.com/stellar/stellar-protocol/issues/146).
+A `rejected` response will have a `400` HTTP status code and `rejected` as the `status` value.
+
+Parameters:
+
+Name | Type | Description
+-----|------|------------
+`status` | string | `"rejected"`
+`error` | string | A human readable string explaining why the transaction is not compliant and could not be made compliant.
+
+###### Example
+
+```json
+{
+  "status": "rejected",
+  "message": "The destination account is blocked."
+}
+```
+
+### Best practices
+
+- Using the `revised` status to compose a sandwich transaction for authorization purpose is encouraged as it can take the burdens away from the wallets.
```

Comment:
I'm not sure about this. Ideally a transaction _isn't_ revised so the user doesn't have to manually inspect the changes. Are the sandwich operations not deterministic? Meaning, can a wallet determine what the wrapper operations need to be depending on what the actual "meat" of the operation does?